wordpress-mobile/WordPress-Android
# Fixes #10108: Add auto save conflict dialog

Fixes #10108: Add and auto save conflict dialog,  [Demo video](https://bia.is/s/pHEI/conflict-dialog-demo.webm)

Related FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1357


Here is the dialog, with current copy:

![Screenshot_1566489119](https://user-images.githubusercontent.com/40213/63529870-be510580-c505-11e9-8b25-673c2ab84f44.png)

For reference, In Calypso:

<img width="422" alt="screenshot-2019-05-03-at-09 25 19" src="https://user-images.githubusercontent.com/40213/57301854-3a9c1600-70da-11e9-8b51-caace0fe6d33.png">

Note the "Saved draft" is not 100% correct, since the post was published, that's why I didn't keep it.

@osullivanchris what do you think about that? Should we ping editorial directly?

To test:
- Create a post in wpandroid and publish it.
- Edit this post in Calypso (Add "Calypso, Calypso, Calypso" to the content), close the browser tab without saving.
- Refresh the Post list in wpandroid.
- You should see a "Unhandled auto save" label.
- Tap edit, conflict should show up, pick "current version".
- Don't edit the post, tap back.
- "Unhandled auto save" label should remain.
- Tap edit, conflict should show up, pick "more recent version".
- Make sure the post looks like what you modified in Calypso ("Calypso, Calypso, Calypso").
- Modify the post (add "WPAndroid, WPAndroid, WPAndroid"), tap back (this should auto save the post)
- "Unhandled auto save" should be gone, you should see a "Local changes" label.
- Open this post on Calypso, tap "Restore", make sure you see "WPAndroid, WPAndroid, WPAndroid" in the post content.
